### ISSUE TITLE
Duplicate key value violates unique constraint

### ISSUE THREAD

**Issue by KrisPetkov** [linq2db/linq2db#2513]
Hello guys,

When I insert a new row in my table **manually** (without specifying explicitly id) the row is inserted without any problems and an id is assigned to this row (automatically incremented). After this if I call a method from my code to insert a row in the same table with this function: `dataContext.InsertWithInt32Identity(entity);` I'm getting the following error:

`Npgsql.PostgresException (0x80004005): 23505: duplicate key value violates unique constraint "PK_Address"`

And if I call my method one more time the insertion is successful. So i think that the insertion method from my code is lacking behind with one id incrementation. 

My table is created as follows:

```sql 
CREATE TABLE public."Address" (
	"Id" int4 NOT NULL GENERATED BY DEFAULT AS IDENTITY,	
	"Address1" varchar(500) NULL,
	"Address2" varchar(500) NULL,
	"ZipPostalCode" varchar(255) NULL,
	"PhoneNumber" varchar(255) NULL,	
	CONSTRAINT "PK_Address" PRIMARY KEY ("Id")
);
```

Thank you in advance! 

### Environment details
linq2db version: 2.9.8
Npgsql version: 4.1.3.1
PostgreSQL version: 12.2
Operating system: Windows 10
.NET Core: 3.1

**Comment by sdanyliv** [comment]
@KrisPetkov, 2.9.8 is no longer supported. Could you please check in 3.1.3?

**Comment by KrisPetkov** [comment]
@sdanyliv You were right it seems that this issue was fixed in some of the updates after 2.9.8 version. After updating to the newest 3.1.3 version I cannot reproduce this issue. Thank you!
I will close the issue.
